File: android/src/main/java/com/squareetlabs/capacitor/nearbymultipeer/BleLogger.java
package com.squareetlabs.capacitor.nearbymultipeer;

import android.util.Log;

/**
 * Sistema de logging mejorado para operaciones BLE
 */
public class BleLogger {
    private static final String TAG = "NearbyMultipeer";
    private static LogLevel logLevel = LogLevel.INFO;
    
    /**
     * Niveles de log
     */
    public enum LogLevel {
        NONE(0),   // No logging
        ERROR(1),  // Solo errores
        WARN(2),   // Errores y advertencias
        INFO(3),   // Errores, advertencias e info general
        DEBUG(4),  // Información detallada
        VERBOSE(5); // Información muy detallada
        
        private final int level;
        
        LogLevel(int level) {
            this.level = level;
        }
        
        public int getLevel() {
            return level;
        }
    }
    
    /**
     * Establece el nivel de log
     * @param level Nivel de log
     */
    public static void setLogLevel(LogLevel level) {
        logLevel = level;
    }
    
    /**
     * Log de error
     * @param message Mensaje
     */
    public static void error(String message) {
        if (logLevel.getLevel() >= LogLevel.ERROR.getLevel()) {
            Log.e(TAG, message);
        }
    }
    
    /**
     * Log de error con excepción
     * @param message Mensaje
     * @param throwable Excepción
     */
    public static void error(String message, Throwable throwable) {
        if (logLevel.getLevel() >= LogLevel.ERROR.getLevel()) {
            Log.e(TAG, message, throwable);
        }
    }
    
    /**
     * Log de advertencia
     * @param message Mensaje
     */
    public static void warn(String message) {
        if (logLevel.getLevel() >= LogLevel.WARN.getLevel()) {
            Log.w(TAG, message);
        }
    }
    
    /**
     * Log de info
     * @param message Mensaje
     */
    public static void info(String message) {
        if (logLevel.getLevel() >= LogLevel.INFO.getLevel()) {
            Log.i(TAG, message);
        }
    }
    
    /**
     * Log de debug
     * @param message Mensaje
     */
    public static void debug(String message) {
        if (logLevel.getLevel() >= LogLevel.DEBUG.getLevel()) {
            Log.d(TAG, message);
        }
    }
    
    /**
     * Log verbose
     * @param message Mensaje
     */
    public static void verbose(String message) {
        if (logLevel.getLevel() >= LogLevel.VERBOSE.getLevel()) {
            Log.v(TAG, message);
        }
    }
    
    /**
     * Log específico para operaciones BLE
     * @param operation Operación
     * @param deviceId ID del dispositivo
     * @param details Detalles
     */
    public static void logBleOperation(String operation, String deviceId, String details) {
        debug(String.format("BLE %s - Device: %s - %s", operation, deviceId != null ? deviceId : "N/A", details));
    }
    
    /**
     * Log para datos hexadecimales (útil para depurar payloads)
     * @param label Etiqueta
     * @param data Datos
     */
    public static void logHexData(String label, byte[] data) {
        if (logLevel.getLevel() >= LogLevel.DEBUG.getLevel() && data != null) {
            StringBuilder hex = new StringBuilder();
            for (byte b : data) {
                hex.append(String.format("%02X ", b));
            }
            debug(label + ": " + hex.toString());
        }
    }
} 

File: ios/Sources/NearbyMultipeerPlugin/BleLogger.swift
import Foundation

/**
 * Sistema de logging mejorado para operaciones BLE
 */
class BleLogger {
    /// Niveles de log
    enum LogLevel: Int {
        case none = 0    // No logging
        case error = 1   // Solo errores
        case warn = 2    // Errores y advertencias
        case info = 3    // Errores, advertencias e info general
        case debug = 4   // Información detallada
        case verbose = 5 // Información muy detallada
    }
    
    /// Tag para los logs
    private static let tag = "NearbyMultipeer"
    
    /// Nivel de log actual
    static var logLevel: LogLevel = .info
    
    /// Habilitar/deshabilitar colores en los logs de consola
    static var enableColors = true
    
    /**
     * Log de error
     * - Parameter message: Mensaje
     */
    static func error(_ message: String) {
        if logLevel.rawValue >= LogLevel.error.rawValue {
            let formattedMessage = enableColors ? "\u{001B}[31m[\(tag)] ❌ \(message)\u{001B}[0m" : "[\(tag)] ❌ \(message)"
            NSLog(formattedMessage)
        }
    }
    
    /**
     * Log de error con errores nativos
     * - Parameters:
     *   - message: Mensaje
     *   - error: Error
     */
    static func error(_ message: String, error: Error) {
        if logLevel.rawValue >= LogLevel.error.rawValue {
            let errorDescription: String
            
            if let bleError = error as? BleError {
                errorDescription = bleError.localizedDescription
            } else {
                errorDescription = error.localizedDescription
            }
            
            let formattedMessage = enableColors ? "\u{001B}[31m[\(tag)] ❌ \(message): \(errorDescription)\u{001B}[0m" : "[\(tag)] ❌ \(message): \(errorDescription)"
            NSLog(formattedMessage)
        }
    }
    
    /**
     * Log de advertencia
     * - Parameter message: Mensaje
     */
    static func warn(_ message: String) {
        if logLevel.rawValue >= LogLevel.warn.rawValue {
            let formattedMessage = enableColors ? "\u{001B}[33m[\(tag)] ⚠️ \(message)\u{001B}[0m" : "[\(tag)] ⚠️ \(message)"
            NSLog(formattedMessage)
        }
    }
    
    /**
     * Log de info
     * - Parameter message: Mensaje
     */
    static func info(_ message: String) {
        if logLevel.rawValue >= LogLevel.info.rawValue {
            let formattedMessage = enableColors ? "\u{001B}[36m[\(tag)] ℹ️ \(message)\u{001B}[0m" : "[\(tag)] ℹ️ \(message)"
            NSLog(formattedMessage)
        }
    }
    
    /**
     * Log de debug
     * - Parameter message: Mensaje
     */
    static func debug(_ message: String) {
        if logLevel.rawValue >= LogLevel.debug.rawValue {
            let formattedMessage = enableColors ? "\u{001B}[32m[\(tag)] 🔍 \(message)\u{001B}[0m" : "[\(tag)] 🔍 \(message)"
            NSLog(formattedMessage)
        }
    }
    
    /**
     * Log verbose
     * - Parameter message: Mensaje
     */
    static func verbose(_ message: String) {
        if logLevel.rawValue >= LogLevel.verbose.rawValue {
            let formattedMessage = enableColors ? "\u{001B}[37m[\(tag)] 📝 \(message)\u{001B}[0m" : "[\(tag)] 📝 \(message)"
            NSLog(formattedMessage)
        }
    }
    
    /**
     * Log específico para operaciones BLE
     * - Parameters:
     *   - operation: Operación
     *   - deviceId: ID del dispositivo
     *   - details: Detalles
     */
    static func logBleOperation(_ operation: String, deviceId: String?, details: String) {
        debug("BLE \(operation) - Device: \(deviceId ?? "N/A") - \(details)")
    }
    
    /**
     * Log para datos hexadecimales (útil para depurar payloads)
     * - Parameters:
     *   - label: Etiqueta
     *   - data: Datos
     */
    static func logHexData(_ label: String, data: Data?) {
        if logLevel.rawValue >= LogLevel.debug.rawValue, let data = data {
            let hexString = data.map { String(format: "%02X ", $0) }.joined()
            debug("\(label): \(hexString)")
        }
    }
} 